src/test/testing_assert.cpp

```diff
@@ -188,4 +188,20 @@ void ASSERT_INNER_JOIN_NODE(const std::shared_ptr<AbstractASTNode>& node, ScanTy
 }
 
 void ASSERT_CROSS_JOIN_NODE(const std::shared_ptr<AbstractASTNode>& node) {}
+
+bool check_ast_tie(const std::shared_ptr<const AbstractASTNode>& parent, ASTChildSide child_side,
```

Comment:
How is this method different from `child->get_child_side(parent) == child_side`?

Reply:
`child->get_child_side(parent)` will crash if `child` is not a child of `parent` which I want to avoid in tests. But well spotted!